### ISSUE TITLE
Seek problem, MediaCodecVideoDecoder: this should never happen

### ISSUE THREAD

**Issue by xylik** [protyposis/ITEC-MediaPlayer#28]
Hi, I've noticed problem with exact seeking on bunch of mp4 files that I've cut with ffmpeg. Am using MediaPlayer-Extended with TextureView.
When I try to seek on video the first time, player seeks  to wrong position and logs this error:
"MediaCodecVideoDecoder: this should never happen"
if I try to seek one more time after the first attempt to exact same time as before, player seeks to correct location.
On this [link](https://drive.google.com/drive/folders/0Byn0xtAvVbCWbThTVWJfSHN1aG8) you can found problematic mp4 file, the file name is: "371_ffmpeg_cut.mp4"
Am trying to seek on that video on 1381ms. Here is logcat output:

D/MediaPlayer: seekTo 1381000
 D/MediaCodecVideoDecoder: seeking to:                 1381000
 D/MediaCodecVideoDecoder: extractor current position: 208541
 D/MediaCodecVideoDecoder: extractor new position:     1369977
 D/MediaCodecVideoDecoder: frame new position:         2502500
 D/MediaCodecVideoDecoder: seeking finished, skipped 0 frames
 W/MediaCodecVideoDecoder: this should never happen
 D/MediaPlayer: onSeekComplete
 D/MediaPlayer: seekTo 1381000
 D/MediaCodecVideoDecoder: seeking to:                 1381000
 D/MediaCodecVideoDecoder: extractor current position: 2752750
 D/MediaCodecVideoDecoder: extractor new position:     0
 D/MediaCodecVideoDecoder: skipping frames...
 D/MediaCodecVideoDecoder: frame new position:         1418083
 D/MediaCodecVideoDecoder: seeking finished, skipped 34 frames
 D/MediaCodecVideoDecoder: exact seek: repeat seek for previous frame at 1376375
 D/MediaCodecVideoDecoder: seeking to:                 1376375
 D/MediaCodecVideoDecoder: extractor current position: 1626625
 D/MediaCodecVideoDecoder: extractor new position:     0
 D/MediaCodecVideoDecoder: skipping frames...
 D/MediaCodecVideoDecoder: frame new position:         1376375
 D/MediaCodecVideoDecoder: seeking finished, skipped 33 frames
 D/MediaCodecVideoDecoder: exact seek match!
 D/MediaPlayer: onSeekComplete

**Comment by protyposis** [comment]
Thank you, you managed to trigger an edge case that "should never happen" :) I cannot reproduce the problem with your test video though. Can you provide code for a minimum working example or instructions on how to demonstrate the problem in the MediaPlayer-Extended demo app?

**Comment by xylik** [comment]
Hi, thx for such a quick reply. Here is [link](https://github.com/xylik/MediaPlayer_SeekProblem/tree/minimal_example) to my example, its quite minimal = one activity + one wrapper class arround media player. Wrapper class is almost identical to  android VideoView implementation...

For example to work, copy 371_ffmpeg_cut.mp4 to this dir:
Environment.getExternalStorageDirectory() + "/FFmpegVideos"

External storage dir is typically root of sdcard or emulated storage which is shown when you connect your phone to your pc with USB, you can copy the file to your phone with file file-explorer...

If you'll have any problem with starting the example am here to assist.

**Comment by protyposis** [comment]
I tried the example on my phone (Nexus 5X, Android 7.0). I start the app and just seek to 1381 ms and it works as expected:

```
D/MediaPlayer: seekTo 1381000
D/MediaCodecVideoDecoder: seeking to:                 1381000
D/MediaCodecVideoDecoder: extractor current position: 250250
D/MediaCodecVideoDecoder: extractor new position:     0
D/MediaCodecVideoDecoder: skipping frames...
D/MediaCodecVideoDecoder: frame new position:         1418083
D/MediaCodecVideoDecoder: seeking finished, skipped 34 frames
D/MediaCodecVideoDecoder: exact seek: repeat seek for previous frame at 1376375
D/MediaCodecVideoDecoder: seeking to:                 1376375
D/MediaCodecVideoDecoder: extractor current position: 1626625
D/MediaCodecVideoDecoder: extractor new position:     0
D/MediaCodecVideoDecoder: skipping frames...
D/MediaCodecVideoDecoder: frame new position:         1376375
D/MediaCodecVideoDecoder: seeking finished, skipped 33 frames
D/MediaCodecVideoDecoder: exact seek match!
D/MediaPlayer: onSeekComplete
```

Looking at your code, I can't see any issues. So I guess this must be related to the system's MediaCodec/Extractor implementation. Did you try it on different devices? Which Android version are you using?

I'm not a fan of workarounds but as a temporary solution you could issue a second seek if the first one fails. The first seek fails instantly and does not decode any frames, so it doesn't really introduce any overhead.

**Comment by xylik** [comment]
The problem is manifested on my Xiaomi Redmi 3 Pro(Lollipop 5.1.1). Currently cant test it on another Lollipop phone but will do that and update this comment.  
I've tested example on two Marshmallow and one KitKat phone  and everything is working fine on them. As you sad example is also working on Nuggat. Thx for pointing me in right direction and keep up with good work on player.

UPDATE:
I've tested example on Motorola G(Lollipop 5.1), the example is working.
